core/modules/views/plugins/views_plugin_style_grid.inc

```diff
@@ -31,41 +35,111 @@ class views_plugin_style_grid extends views_plugin_style {
+    $options = views_grid_get_options($this->options);
+
     $form['columns'] = array(
-      '#type' => 'number',
+      '#type' => 'textfield',
       '#title' => t('Number of columns'),
-      '#default_value' => $this->options['columns'],
+      '#default_value' => $options['columns'],
       '#required' => TRUE,
-      '#min' => 1,
-      '#step' => 1,
+      '#element_validate' => array('views_element_validate_integer'),
+    );
+    $form['automatic_width'] = array(
+      '#type' => 'checkbox',
+     '#title' => t('Automatic width'),
+     '#description' => t('The width of each column will be calculated automatically based on the number of columns entered.'),
+     '#default_value' => $options['automatic_width'],
     );
     $form['alignment'] = array(
       '#type' => 'radios',
       '#title' => t('Alignment'),
       '#options' => array('horizontal' => t('Horizontal'), 'vertical' => t('Vertical')),
-      '#default_value' => $this->options['alignment'],
+      '#default_value' => $options['alignment'],
       '#description' => t('Horizontal alignment will place items starting in the upper left and moving right. Vertical alignment will place items starting in the upper left and moving down.'),
     );
-
-    $form['fill_single_line'] = array(
+    $form['default_classes'] = array(
       '#type' => 'checkbox',
-      '#title' => t('Fill up single line'),
-      '#description' => t('If you disable this option, a grid with only one row will have the same number of table cells (<TD>) as items. Disabling it can cause problems with your CSS.'),
-      '#default_value' => !empty($this->options['fill_single_line']),
+      '#title' => t('Remove default views classes'),
+      '#default_value' => $options['default_classes'],
     );
-
-    $form['caption'] = array(
+    $form['grid']= array(
+      '#type' => 'fieldset',
+      '#title' => t('Grid Classes'),
+      '#weight' => 70,
+      '#collapsible' => TRUE,
+      '#collapsed' => TRUE,
+    );
+    $form['wrapper_classes'] = array(
       '#type' => 'textfield',
-      '#title' => t('Short description of table'),
-      '#description' => t('Include a caption for better accessibility of your table.'),
-      '#default_value' => $this->options['caption'],
+      '#title' => t('Wrapper'),
+      '#default_value' => $options['wrapper_classes'],
+      '#fieldset' => 'grid',
+      '#prefix' => '<div class="views-left-30">',
+      '#suffix' => '</div>',
     );
-
-    $form['summary'] = array(
+    $form['column_class_wrapper']= array(
+      '#type' => 'fieldset',
+      '#title' => t('Column Classes'),
+      '#weight' => 80,
+      '#collapsible' => TRUE,
+      '#collapsed' => TRUE,
+    );
+    $form['column_classes'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Every Column'),
+      '#default_value' => $options['column_classes'],
+      '#fieldset' => 'column_class_wrapper',
+      '#prefix' => '<div class="views-left-30">',
```

Comment:
Thinking it may be worth it to write some CSS for this, the views UI CSS is just abysmal. I can help with that.